### PR TITLE
Allow Chevron click on select button

### PR DESCRIPTION
**Pull request by iethree**
## Description

The `SelectButton` component conditionally renders either a chevron or close icon, depending on the state of the component. However, the icon itself has a click action which was conditional only upon whether an `onClear` function was passed. This updates that logic to only use the `onClear` function when the `close` icon appears, keeping the `chevronDown` icon from triggering the `onClear` function, and allowing it to bubble up to the main `onClick` handler.

I also added in the `onClick` prop explicitly, to reduce the number of inscrutable `...rest` props.

**tl;dr: clicking on the chevron icon works as expected now**

Before | After
--- | ---
clickfail | clickyes

### Testing steps

- open the bulk filter modal on any table
- click a chevron in a select box, and see that the popover opens!

### CODE REVIEW

**Review by WiNloSt — CHANGES_REQUESTED** [review]
After fixing the type, you should be good. Also, I just notice one E2E failed, that doesn't have anything to do with this PR, but I've issued the fix https://github.com/metabase/metabase/pull/23348. After you rebased off `master` after the fix lands then the test should pass.

**Review by WiNloSt — APPROVED** [review]
